src/openforms/registrations/tests/test_views.py

```diff
@@ -170,19 +95,101 @@ def test_missing_objects_api_group(self, m):
 
         self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
 
-    def test_retrieve_with_explicit_objects_api_group(self, m):
+    def test_retrieve_with_explicit_objects_api_group(self):
         user = StaffUserFactory.create()
         url = furl(reverse("api:iotypen-list"))
-        url.args["objects_api_group"] = self.objects_api_group1.pk
+        url.args["objects_api_group"] = self.objects_api_group.pk
         url.args["registration_backend"] = "objects_api"
         self.client.force_login(user)
 
-        self.install_mocks(m)
+        response = self.client.get(url.url)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        data = response.json()
+
+        self.assertEqual(len(data), 3)
+
+    def test_retrieve_filter_by_catalogus_missing_param(self):
+        user = StaffUserFactory.create()
+        url = furl(reverse("api:iotypen-list"))
+        url.args["objects_api_group"] = self.objects_api_group.pk
+        url.args["registration_backend"] = "objects_api"
+        url.args["catalogus_domein"] = "TEST"
+        # We don't add `catalogus_rsin` here
+        self.client.force_login(user)
+
+        response = self.client.get(url.url)
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_retrieve_filter_by_catalogus(self):
+        user = StaffUserFactory.create()
+        url = furl(reverse("api:iotypen-list"))
+        url.args["objects_api_group"] = self.objects_api_group.pk
+        url.args["registration_backend"] = "objects_api"
+        url.args["catalogus_domein"] = "TEST"
+        url.args["catalogus_rsin"] = "000000000"
```

Comment:
please use the simple django API where possible, `furl` should only be used where we have to process incoming URL strings ourselves or when doing complex URL mutations. The much simpler equivalent is:

```py
response = self.client.get(
    reverse(""api:iotypen-list"),
    {
        "objects_api_group": self.objects_api_group.pk,
        "registration_backend": "objects_api",
        "catalogus_domein": "TEST",
        "catalogus_rsin": "000000000",
    },
)
```